[SEP-0024] Add Soroban contract account and SEP-0045 support (#1783)
stellar/stellar-protocol: ecosystem/sep-0024.md

- An anchor must define the location of their `TRANSFER_SERVER_SEP0024` in
  their [`stellar.toml`](sep-0001.md). This is how a wallet knows where to find
  the anchor's server.

> - Anchors and clients must support [SEP-10](sep-0010.md) or

Author: It should be `or` because they MUST do one of them.

> - Anchors and clients must support [SEP-10](sep-0010.md) or

Reviewer: but they can support both right?

> - Anchors and clients must support [SEP-10](sep-0010.md) or

Author: Yes, they can. But when we say they must support SEP-10 and SEP-45, that means they cannot just support SEP-10 or SEP-45.

### Shared, Omnibus, or Pooled Accounts

Client applications can use a single classic Stellar account to hold multiple
users' funds. To distinguish users that use the same Stellar account, the

> decoded SEP-10 JWT's `sub` value may contain a memo value after the Stellar

Author: The account of SEP-10 is always classic. There are quite a few places like this. If the context is obvious, I think we may omit the `classic`.

> decoded SEP-10 JWT's `sub` value may contain a memo value after the Stellar

Author: I think classic is a relative term. It is more accurate to say "non-contract" account or G-account. In the context of SEP-10, it is clear. 

But if we want to make clear of the account everywhere, it can be very verbose. I think within the SEP readers' context, the account should mean classic if not specified.

> decoded SEP-10 JWT's `sub` value may contain a memo value after the Stellar

Author: The definition of the [classic account ](https://github.com/stellar/stellar-protocol/pull/1783/files#diff-8eff781a81bdbf7d9a0f55d126cde3920a205a82b92ce13156269d725a50383bR56)is also added to avoid confusion.

| `amount`                      | number                                  | (optional) Amount of asset requested to deposit. If this is not provided it will be collected in the interactive flow. When `qoute_id` is specified, this parameter must match the quote's `quote.sell_amount` or be omitted.                                                                                                                                                                                                                                                             |
| `quote_id`                    | string                                  | (optional) The `id` returned from a `SEP-38 POST /quote` response. If this parameter is provided and the user delivers the deposit funds to the Anchor before the quote expiration, the Anchor must respect the conversion rate agreed in that quote. When `quote_id` is set, `asset_code`, `source_asset` and `amount` must be validated by the anchor, if present. In case of a conflict with the ones used to create the [SEP-38] quote, this request should be rejected with a `400`. |

> | `account`                     | `G...` or `M...` or `C...` string       | (optional) The classic account, contract account or muxed account the client wants to use as the destination of the payment sent by the anchor. Defaults to the account authenticated via SEP-10 or SEP-45 if not specified.                                                                                                                                                                                                                                                              |

Reviewer: l
```suggestion
| `account`                     | `G...` or `M...` or `C...` string       | (optional) The classic account, contract account or muxed account the client wants to use as the destination of the payment sent by the anchor. Defaults to the account used in authentication if not specified.                                                                                                                                                                                                                                                              |
```

| `amount`            | number                                  | (optional) Amount of asset requested to withdraw. If this is not provided it will be collected in the interactive flow. When `qoute_id` is specified, this parameter must match the quote's `quote.sell_amount` or be omitted.                                                                                                                                                                                                                                                                 |
| `quote_id`          | string                                  | (optional) The `id` returned from a `SEP-38 POST /quote` response. If this parameter is provided and the user delivers the deposit funds to the Anchor before the quote expiration, the Anchor must respect the conversion rate agreed in that quote. When `quote_id` is set, `asset_code`, `destination_asset` and `amount` must be validated by the anchor, if present. In case of a conflict with the ones used to create the [SEP-38] quote, this request should be rejected with a `400`. |

> | `account`           | `G...` or `M...` or `C...` string       | (optional) The class account, contract account, or muxed account the client will use as the source of the withdrawal payment to the anchor. Defaults to the account authenticated via SEP-10 or SEP-45 if not specified.                                                                                                                                                                                                                                                                       |

Reviewer: ```suggestion
| `account`           | `G...` or `M...` or `C...` string       | (optional) The class account, contract account, or muxed account the client will use as the source of the withdrawal payment to the anchor. Defaults to the account used in authentication if not specified.                                                                                                                                                                                                                                                                       |
```

The transaction endpoint enables clients to query/validate a specific
transaction at an anchor.

> Anchors must ensure that the SEP-10 or SEP-45 JWT included in the request
> contains the Stellar account and optional memo value used when making the
> original deposit or withdraw request that resulted in the transaction requested
> using this endpoint.

Reviewer: ```suggestion
Anchors must verify that the SEP-10 or SEP-45 JWT in the request contains the Stellar account and, if applicable, the memo that were used in the original deposit or withdrawal request for the transaction being queried.
```
This was a really loooooong sentence...
